lib/internal/util/inspect.js

```diff
@@ -1305,13 +1305,122 @@ function identicalSequenceRange(a, b) {
           len++;
         }
         if (len > 3) {
-          return { len, offset: i };
+          return [len, i];
         }
       }
     }
   }
 
-  return { len: 0, offset: 0 };
+  return [0, 0];
+}
+
+function getDuplicateErrorFrameRanges(frames) {
+  // Build a map: frame line -> sorted list of indices where it occurs
+  const result = [];
+  const lineToPositions = new SafeMap();
+
+  for (let i = 0; i < frames.length; i++) {
+    const positions = lineToPositions.get(frames[i]);
+    if (positions === undefined) {
+      lineToPositions.set(frames[i], [i]);
+    } else {
+      positions.push(i);
```

Comment:
```suggestion
      positions[positions.length] = i;
```

Reply:
you marked this as resolved; is `.push` faster than my suggestion?

Reply:
true - but this avoids a primordial either being needed, or being missing :-)

Reply:
I just changed it